EssentialsDiscord/src/main/resources/config.yml

```
@@ -0,0 +1,292 @@
+#
+# Firstly, channels defined here can be used to give players permission to see messages from said channel.
+# This can be done by give your players the permission node `essentials.discord.receive.<channel>`.
+# For example, if you wanted to let a player see messages from the primary channel, you'd give them `essentials.discord.receive.primary`.
+#
+# Secondly, channels defined here can be used in the section below to specify which channel a message goes to.
+# If a defined channel id is invalid, the primary channel will be used as a fallback.
+# If the primary channel is not defined or invalid, the default channel of the server will be used.
+# If your server doesn't have any text channels, the plugin will be disabled.
+# By default, two channels are defined; (You will need to replace the zeros with the actual channel id you want to use)
+# - primary, which will send basic join/leave/death/chat messages
+# - staff, which will send kick/mute messages
+channels:
+  primary: 000000000000000000
+  staff: 000000000000000000
+
+# Should all players receive discord messages from the primary channel, regardless of their permissions?
+# This is intended for use for people without permissions plugins. If you have a permissions plugin, please give your
+# players the essentials.discord.receive.primary permission.
+always-receive-primary: false
+
+# Chat relay settings
+# General settings for discord to mc and mc to discord chat relays.
+# To configure the channel minecraft chat is sent to, see the "message-types" section of the config.
+chat:
+  # The maximum amount of characters messages from discord should be before being truncated.
+  discord-max-length: 2000
+  # Whether or not new lines from discord should be filtered or not.
+  filter-newlines: true
+  # A regex pattern which will not send matching messages through to discord.
+  # By default, this will ignore messages starting with '!' and '?'.
+  discord-filter: "^[!?]"
+  # Whether or not to show all mc chat messages that are not shown to all players.
+  # You should only need to enable this if you're not seeing all chat messages go through to discord.
+  show-all-chat: false
+
+# Console relay settings
+# The console relay will send every message that is sent to the console, to a discord channel.
+console:
+  # The channel id (or webhook url) the console output should be sent to.
+  # If the channel id/webhook url is invalid or set to 'none', the console relay will be disabled.
+  # Note: If you use a channel id the bot needs the "Manage Webhooks" permission in discord or else it will not work.
+  channel: 000000000000000000
+  # The format of the console message.
+  # The following placeholders can be used;
+  # {timestamp} - Timestamp in HH:mm:ss format
+  # {level} - The console logging level
+  # {message} - The actual message being logged
+  format: "[{timestamp} {level}] {message}"
+  # The username the console logger webhook will make.
```

```suggestion
  # The name of the webhook that will be created, if a channel ID is provided.
```

I think this is what it does?